# Description: Get the correct device file for SANE scanners.
# Ubuntu: https://bugs.launchpad.net/bugs/36137
# Upstream: http://bugzilla.gnome.org/show_bug.cgi?id=525353
--- gnome-volume-manager-2.22.1/src/manager.c	2008-03-31 17:06:53.000000000 +0200
+++ gnome-volume-manager-2.22.1.new/src/manager.c	2008-03-31 17:07:50.000000000 +0200
@@ -1222,10 +1222,18 @@
 	
 	dbus_error_init (&error);
 	if (!(device = libhal_device_get_property_string (hal_ctx, udi, "scanner.device", &error))) {
-		warn ("cannot get scanner.device property: %s", error.message);
 		if (dbus_error_is_set (&error))
 			dbus_error_free (&error);
-		return;
+                /* check parent's linux.device_file */
+                device = libhal_device_get_property_string(hal_ctx, udi, "info.parent", NULL);
+                if (device) {
+                        device = libhal_device_get_property_string (hal_ctx, device, 
+                                "linux.device_file", &error);
+                }
+                if (!device) {
+                        warn ("cannot get scanner.device property: %s", error.message);
+                        return;
+                }
 	}
 	
 	gvm_run_command (config.autoscanner_command, udi, device, NULL);
